packages/lesswrong/components/editor/EditorFormComponent.tsx

```diff
@@ -435,6 +454,24 @@ export const EditorFormComponent = ({
 
   const actualPlaceholder = ((collectionName === "Posts" && getPostPlaceholder(document)) || editorHintText || hintText || placeholder);
 
+  useEffect(() => {
+    if (!isCollabEditor && collectionName === 'Posts' && fieldName === 'contents') {
+      setAutosaveEditorState((_) => () => new Promise((resolve, reject) => {
+        if (editorRef.current && shouldSubmitContents(editorRef.current)) {
+          void editorRef.current?.submitData()
+            .then(({ originalContents: { type, data: value }}) => ({ type, value }))
+            .then(saveRemoteBackup)
+            .then(resolve)
+            .catch(reject);
+        }
+      }));
+    }
+
+    return () => {
+      setAutosaveEditorState(null);
+    }
+  }, [isCollabEditor, collectionName, fieldName, saveRemoteBackup, setAutosaveEditorState]);
```

Comment:
This `useEffect` makes me nervous because it sets state via a context provider in a way that will cause this component to rerender, and it's not entirely straightorward to verify that this doesn't turn into a rerender-loop via `saveRemoveBackup` or similar.

Reply:
Hm... calling `setAutosaveEditorState` causes `EditorFormComponent` to rerender because `autosaveEditorState` gets changed, and even though `EditorFormComponent` isn't using it, it's still using the context which contains it, and any change to that context causes a rerender?